go/pkg/sysdb/metastore/db/dao/collection.go

```diff
@@ -459,3 +459,58 @@ func (s *collectionDb) UpdateVersionRelatedFields(collectionID, existingVersionF
 	}
 	return result.RowsAffected, nil
 }
+
+func (s *collectionDb) LockCollection(collectionID *string) error {
```

Comment:
note: GC has Optimistic Concurrency Control, should do select for update there as well